app/controllers/users/registrations_controller.rb

```diff
@@ -9,5 +16,15 @@ def update
         respond_with resource
       end
     end
+
+    private
+
+    def their_own?
+      current_user.id.eql?(params[:id].to_i)
+    end
+
+    def authorize_for_current_user
+      authorize current_user if their_own?
```

Comment:
The idea is to allow currently logged in users to edit their own passwords only.